T/Torch/Torch/build_tarballs.jl

```diff
@@ -0,0 +1,111 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Torch"
+version = v"1.10.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/pytorch/pytorch.git", "71f889c7d265b9636b93ede9d651c0a9c4bee191"),
+    FileSource("https://micromamba.snakepit.net/api/micromamba/linux-64/0.21.1", "c907423887b43bec4e8b24f17471262c8087b7095683f41dcef4a4e24e9a3bbd"; filename = "micromamba.tar.bz2"),
+    ArchiveSource("https://github.com/JuliaBinaryWrappers/CUDA_full_jll.jl/releases/download/CUDA_full-v10.2.89%2B5/CUDA_full.v10.2.89.x86_64-linux-gnu.tar.gz", "60e6f614db3b66d955b7e6aa02406765e874ff475c69e2b4a04eb95ba65e4f3b"; unpack_target = "CUDA_full.v10.2"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+
+mkdir micromamba
+cd micromamba
+tar xfj ../micromamba.tar.bz2
+export PATH=$PATH:$WORKSPACE/srcdir/micromamba/bin
+./bin/micromamba shell init -s bash -p ~/micromamba
+source ~/.bashrc
+micromamba activate
+micromamba install -y python=3.9 pyyaml typing_extensions mkl-include -c conda-forge
+
+cd $WORKSPACE/srcdir/pytorch
+
+if [[ $bb_full_target == *cuda* ]]; then
+    cuda_version=`echo $bb_full_target | sed -E -e 's/.*cuda\+([0-9]+\.[0-9]+).*/\1/'`
+    cuda_version_major=`echo $cuda_version | cut -d . -f 1`
+    cuda_version_minor=`echo $cuda_version | cut -d . -f 2`
+    cuda_full_path="$WORKSPACE/srcdir/CUDA_full.v$cuda_version/cuda"
+    apk del cmake
+    apk add 'cmake<3.17' --repository=http://dl-cdn.alpinelinux.org/alpine/v3.11/main
```

Comment:
Wait, what?

Reply:
I got fed up trying to hack the "find cuda" part of Torch - downgrading cmake seems to work.

Torch seems to be using the "old" https://cmake.org/cmake/help/latest/module/FindCUDA.html approach which was replaced with https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html in CMake v3.17 as I understand it.

Reply:
Sigh

Reply:
At least it seems more "clean" to me than patching some set of cmake files :-)

Reply:
... and then it works in the BB shell (e.g. with `--debug=begin`), but not in auto mode...

Reply:
> ... and then it works in the BB shell (e.g. with --debug=begin), but not in auto mode...

I'm not 100% sure what you mean here, but note that package installation and setting environment variables aren't persistent at the moment, when dropping into the debug shell (package are installed in a tmpfs which is lost when recreating the debug environment, and we don't have a way to remember all environment settings). It may be possible to address some of these issues in the future, but at the moment you have to repeat those operations manually